oci/layout/oci_transport.go

```diff
@@ -72,11 +73,11 @@ func ParseReference(reference string) (types.ImageReference, error) {
 	return NewReference(dir, image)
 }
 
-// NewReference returns an OCI reference for a directory and a image.
+// NewReferenceWithIndex returns an OCI reference for a directory and a image， sourceIndex points to an image.
 //
 // We do not expose an API supplying the resolvedDir; we could, but recomputing it
 // is generally cheap enough that we prefer being confident about the properties of resolvedDir.
-func NewReference(dir, image string) (types.ImageReference, error) {
+func NewReferenceWithIndex(dir, image string, sourceIndex int) (types.ImageReference, error) {
```

Comment:
Please document the `sourceIndex == -1` case, and enforce that `sourceIndex` and `image` can’t be set simultaneously.

---

Actually docker/archive has separate `NewReference` and `NewIndexReference`, making it structurally impossible to set both at the same time, and allowing validation that refuses a negative index in the “index” case (OTOH that validation doesn’t exist).